packages/astro/src/core/render/dev/css.ts

```diff
@@ -29,8 +29,8 @@ export function getStylesForURL(filePath: URL, viteServer: vite.ViteDevServer):
 
 		scanned.add(moduleName.id);
 
-		// scan importedModules
-		for (const importedModule of moduleName.importedModules) {
+		// scan importers and importedModules
+		for (const importedModule of [...moduleName.importers, ...moduleName.importedModules]) {
```

Comment:
This solutions just feels wrong 🤔  I tracked the issue down to stylesheets imported with ESM not being included in the module's `importedModules` until **after** HMR triggers a rebuild

In the case of tailwind, it's pretty interesting that the stylesheet is actually listed as an _importer_ of the Astro page, rather than the other way around

Reply:
honestly, this may be a good thing to bring up in the Vite discord, since this is mainly all "Vite /w Postcss" code that they own and maintain, they probably have some insight into what's going on